### DISCUSSION TITLE
"Out of Memory" ("OOM") event

### DISCUSSION THREAD

**Discussion by VinsMediaX** [vercel/community#6445] [Help] [answered]
### Question

Recently, after the site has gone into production, very often the build fails. The error at the end of the build, the error at the end of the build, when it tries to deploy the output is: 

• At least one "Out of Memory" ("OOM") event was detected during the build.

• This occurs when processes or applications running during the build completely fill up the available memory (RAM) in the build container. When this happens, the build container terminates one of the processes during the build with a SIGKILL signal.

I don't know how to solve it. More or less 1 out of 5 builds ends successfully, but very often it fails because of this error.
Sometimes there's another kind of error: "fatal error: all goroutines are asleep - deadlock!". This also I have no idea what it means.

Can you please help me?

<img width="1214" alt="Screenshot 2023-11-17 alle 15 41 52" src="https://github.com/vercel/vercel/assets/99189724/883c0d9b-b765-4050-8e54-278263d9abc9">

**Comment by stephenasuncionDEV** [comment]
![image](https://github.com/vercel/vercel/assets/73958774/3b87ff00-6752-4a2b-a07c-77f7f58b4730)
Having the same issue.

**Comment by stephenasuncionDEV** [comment]
In my case, I solved it by removing one of my components that was being rendered recursively.

**Comment by eby-paul-daniel** [comment]
We're too now facing the similar issue with one of our branches named "development"

**Comment by elie222** [comment]
Same issue here. This commit broke the build:
https://github.com/elie222/inbox-zero/commit/83bcf2a015ee44a8f8ad7845ba31394e62df7f67

Trying to increase memory now to see if that helps: 

```json
"build": "NODE_OPTIONS=--max_old_space_size=2048 prisma migrate deploy && next build",
```

Update: this fixed it.

**Comment by Rupert91** [comment]
Encountering "out of memory" error when deploying Next.js app on Vercel

I'm facing an issue while deploying my Next.js application on Vercel and would greatly appreciate any help or insights from the community.

During the deployment process, the build fails with an "out of memory" error. I suspect there might be some problems within my code causing this issue. Here's the key part of the error log:

bash
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
The complete error stack trace is as follows:

rust
----- Native stack trace -----
 1: 0xb84bd6 node::OOMErrorHandler(char const*, v8::OOMDetails const&) [node]
 2: 0xefead0 v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [node]
 3: 0xefedb7 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, v8::OOMDetails const&) [node]
 ...
Additionally, Vercel's build report shows some large files:

bash
• 4 files larger than 100 MB detected on disk:
  ‣ 159 MB : /vercel/.cache/Cypress/13.6.2/Cypress/Cypress
  ‣ 143 MB : node_modules/@next/swc-linux-x64-musl/next-swc.linux-x64-musl.node
  ‣ 119 MB : node_modules/@next/swc-linux-x64-gnu/next-swc.linux-x64-gnu.node
  ‣ 115 MB : /vercel/.cache/Cypress/13.6.2/Cypress/browser_v8_context_snapshot.bin
I've tried examining my code for potential memory leaks but haven't found any obvious issues yet. I was wondering if anyone has encountered a similar problem or has any suggestions on how to resolve this.

Any help would be greatly appreciated!

**Reply by amyegan** [reply, marked as answer]
Hey @Rupert91. You can explore what's using excessive memory during the build process using theses tips: https://nextjs.org/docs/app/building-your-application/optimizing/memory-usage

**Reply by wscourge** [reply]
This didn't help me at all. Here's what I see now:

```
Error: Command "bash ./scripts/vercel/build.sh" exited with 137
▲ Build system report
• At least one "Out of Memory" ("OOM") event was detected during the build.
  • This occurs when processes or applications running during the build completely fill up the available memory (RAM) in the build container. When this happens, the build container terminates one of the processes during the build with a SIGKILL signal.
  • Read this troubleshooting guide for more information: https://vercel.link/troubleshoot-build-errors
• The build container has plenty of disk space left
• Folder sizes on disk:
  ‣ Input source code:     56 MB
  ‣ Build cache:         6175 MB
  ‣ Output files:           1 MB
  ‣ Node modules:        1239 MB
• 10 files larger than 100 MB detected on disk:
  ‣ 1032 MB : .next/cache/webpack/server-production/10.pack
  ‣ 1012 MB : .next/cache/webpack/client-production/0.pack
  ‣  573 MB : .next/cache/webpack/server-production/4.pack
  ‣  528 MB : .next/cache/webpack/server-production/0.pack
  ‣  276 MB : .next/cache/webpack/client-production/9.pack
  ‣  252 MB : .next/cache/webpack/server-production/9.pack
```

<img width="1339" alt="image" src="https://github.com/vercel/community/assets/16985871/5370a2bd-17d4-48e1-8eb3-c6b4e2b47285">

**Comment by JClackett** [comment]
Also now running into this as soon as I migrated to app router:

```
• The build container has plenty of disk space left
• Folder sizes on disk:
  ‣ Input source code:    352 MB
  ‣ Build cache:         7878 MB
  ‣ Output files:           1 MB
  ‣ Node modules:           1 MB
• 10 files larger than 100 MB detected on disk:
  ‣ 1411 MB : packages/web/.next/cache/webpack/server-production/1.pack
  ‣ 1154 MB : packages/web/.next/cache/webpack/client-production/0.pack
  ‣  495 MB : packages/web/.next/cache/webpack/server-production/11.pack
  ‣  471 MB : packages/web/.next/cache/webpack/server-production/18.pack
  ‣  201 MB : packages/web/.next/cache/webpack/server-production/17.pack
  ‣  166 MB : packages/web/.next/cache/webpack/server-production/0.pack
  ‣  163 MB : packages/web/.next/cache/webpack/client-production/11.pack
  ‣  161 MB : packages/web/.next/cache/webpack/client-production/21.pack
  ‣  160 MB : .git/objects/pack/pack-fad55f2a704a36e07393427a117c214f959dc810.pack
  ‣  150 MB : node_modules/.pnpm/@next+swc-linux-x64-musl@14.2.4/node_modules/@next/swc-linux-x64-musl/next-swc.linux-x64-musl.node
```
